lib/storytime/post_notifier.rb

```diff
@@ -0,0 +1,19 @@
+module Storytime
+  class PostNotifier
+    def self.send_notifications_for(post_id)
+      post = Storytime::Post.find_by(id: post_id)
+
+      return if post.nil?
+
+      unless post.notifications_sent_at
+        post.site.active_email_subscriptions.each do |subscription|
+          mail = Storytime::SubscriptionMailer.new_post_email(post, subscription)
+
+          Rails::VERSION::MINOR < 2 ? mail.deliver : mail.deliver_now
+        end
+
+        post.update_attributes(notifications_sent_at: Time.now)
```

Comment:
Let's move this to just above line 9. If for some reason new_post_email craps out, I'd rather have already set this flag vs. having it try to send the emails for the same post over and over. Emails failing to send is a better error case, to me, than a shit ton of emails accidentally going out.